include/lbann/utils/random.hpp

```diff
@@ -113,6 +113,86 @@ inline T fast_rand_int_pow2(Generator& g, T max) {
   return x & ((typename Generator::result_type) max);
 }
 
+// Methods for quickly generating uniformly random values in [0, 1).
+
+namespace details {
+
+// See section on converting uint64_ts to doubles in:
+// http://xoshiro.di.unimi.it/
+
+template <typename Generator>
+inline float random_float_32(Generator& g) {
+  const uint32_t r = g() >> 9;
+  return r * (1.0f / 8388608.0f);
+}
+
+template <typename Generator>
+inline float random_float_64(Generator& g) {
+  const uint32_t r = uint32_t(g()) >> 9;  // Truncate.
```

Comment:
Is there a reason we can't consolidate `random_float_32` and `random_float_64`? It seems to me that this implementation works fine if `g()` outputs either a `float` or `double`. If we're worried about an unnecessary constructor, switch to brace-initialization.

Reply:
It was mostly just copy-pasting from the `double` code. I don't think it's necessary.